modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -26,189 +24,107 @@ export class Transaction extends BaseTransaction {
     this._signatures.push(signature);
   }
 
+  /**
+   * Replace transaction signatures with the ones from
+   * the inner SDK transaction
+   */
+  private replaceSignatures(): void {
+    this._signatures = this.hederaTx
+      ._toProto()
+      .getSigmap()!
+      .getSigpairList()
+      .map(sigPair => {
+        return toHex(sigPair.getEd25519_asU8());
+      });
+  }
 
   /** @inheritdoc */
   toBroadcastFormat(): string {
-    return toHex(this.encode(this._hederaTx));
+    return toHex(this.hederaTx.toBytes());
   }
 
   /** @inheritdoc */
   toJson(): TxData {
-    const [acc, time] = this.getTxIdParts();
+    const txData = this.hederaTx._toProto().toObject();
+    const txBody = this.txBody.toObject();
+
+    const acc = stringifyAccountId(txBody.transactionid!.accountid!);
+    const time = stringifyTxTime(txBody.transactionid!.transactionvalidstart!);
     const result: TxData = {
       id: acc + '@' + time,
-      hash: this.getTxHash(), // TODO: Update once hedera-sdk release this functionality BGA-284
-      data: toHex(this._hederaTx.bodyBytes),
-      fee: new BigNumber(this._txBody.transactionFee!.toString()).toNumber(),
+      hash: txData.sigmap ? toHex(this.hederaTx.hash()) : '', // Hash is returned if the transaction was signed
+      data: toHex(this.hederaTx._toProto().getBodybytes_asU8()),
+      fee: new BigNumber(txBody.transactionfee).toNumber(),
       from: acc,
       startTime: time,
-      validDuration: this._txBody.transactionValidDuration!.seconds!.toString(),
-      node: stringifyAccountId(this._txBody.nodeAccountID!),
-      memo: this._txBody.memo,
+      validDuration: txBody.transactionvalidduration!.seconds.toString(),
+      node: stringifyAccountId(txBody.nodeaccountid!),
+      memo: txBody.memo,
     };
 
-    if (this._txBody.data === 'cryptoTransfer') {
-      const [recipient, amount] = this.getTransferData();
-      result.amount = amount;
-      result.to = recipient;
+    if (txBody.cryptotransfer) {
+      const transfers = txBody.cryptotransfer!.transfers!.accountamountsList[1];
+      result.amount = transfers.amount;
+      result.to = stringifyAccountId(transfers.accountid!);
     }
     return result;
   }
 
-  /**
-   * Get the recipient account and the amount
-   * transferred on this transaction
-   *
-   * @returns {[string, string]} first element is the recipient, second element is the amount
-   */
-  private getTransferData(): [string, string] {
-    let transferData;
-    this._txBody.cryptoTransfer!.transfers!.accountAmounts!.forEach(transfer => {
-      const amount = Long.fromValue(transfer.amount!);
-      if (amount.isPositive()) {
-        transferData = [stringifyAccountId(transfer.accountID!), amount.toString()];
-      }
-    });
-
-    return transferData;
-  }
-
   //region getters & setters
-  get txBody(): proto.TransactionBody {
-    return this._txBody;
+  get txBody(): TransactionBody {
```

Comment:
I think that this method shouldn't be a getter and we should rename it as another function with some jsDoc